builder/bootloader.py

```diff
@@ -57,34 +57,83 @@ def get_suitable_optiboot_binary(framework_dir, board_config):
 def get_suitable_urboot_binary(framework_dir, board_config):
     mcu = board_config.get("build.mcu", "").lower()
     f_cpu = int(board_config.get("build.f_cpu", "16000000L").strip("UL"))
-    oscillator = board.get("hardware.oscillator", "external").lower()
-    bootloader_led = board_config.get("bootloader.led_pin", "no-led").lower()
+    oscillator = board_config.get("hardware.oscillator", "external").lower()
     bootloader_speed = board_config.get("bootloader.speed", env.subst("$UPLOAD_SPEED"))
-    bootloader_file = "urboot_%s.hex" % mcu
 
     if core == "MicroCore":
+        bootloader_file = "urboot_%s.hex" % mcu
+        bootloader_led = board_config.get("bootloader.led_pin", "no-led").lower()
         f_cpu_error = float(board_config.get("hardware.f_cpu_error", "0.0"))
-        uart = board_config.get("hardware.uart", "swio_rxb1_txb0").lower()
+        uart_pins = board_config.get("hardware.uart", "swio_rxb1_txb0").lower()
```

Comment:
The thing is that the ATtiny13 doesn't have a hardware UART, so its software UART pins have to be specified somehow.
When I implemented PlatformIO Urboot support for the ATtiny13, I just re-used `board_hardware.uart` for the ATtiny13, even though it should _technically_ be `board_hardware.uart_pins`.

What would you prefer? using `board_hardware.uart` for both pins (MicroCore) and the actual UART number (Mega/Mighty/Mini/MajorCore)?

Actually, what I could rather do is to add support for both uart numbers (`uart0`..`uart3`) _and_ uart pins for all cores (`uart0_rxe0_txe1` etc.). That would be the most compatible option I think.
This would mean that it wouldn't matter if you used `board_hardware.uart = uart0` or `board_hardware.uart = uart0_rxd0_txd1` for the ATmega328P.

Reply:
I guess the principle of least astonishment is applicable here, I mean if you're extracting pins, then it should be `board_hardware.uart_pins`, right?

Reply:
Yes, for MicroCore/ATtiny13 (Or any AVR that may implement Urboot and doesn't have a hardware serial port). I should probably add `board_hardware.uart_pins` and promote this in the MicroCore platformio.ini template.
Is this what you'd prefer?

Reply:
> Is this what you'd prefer?

Would be great.